fdbclient/include/fdbclient/StorageCheckpoint.h

```diff
@@ -168,7 +174,20 @@ struct DataMoveMetaData {
 
 	template <class Ar>
 	void serialize(Ar& ar) {
-		serializer(ar, id, version, ranges, priority, src, dest, checkpoints, phase, mode);
+		// 71.2 order serializer(ar, id, version, range, phase, src, dest);
+		if (ar.isDeserializing) {
+			serializer(ar, id, version, range, phase, src, dest, ranges, checkpoints, priority, mode);
+			if (!range.empty() && ranges.empty()) {
+				// upgrade from 71.2 so ranges is empty
+				ranges.push_back(range);
+			}
+		} else {
+			if (ranges.empty()) {
+				serializer(ar, id, version, range, phase, src, dest, ranges, checkpoints, priority, mode);
+			} else {
+				serializer(ar, id, version, ranges[0], phase, src, dest, ranges, checkpoints, priority, mode);
+			}
+		}
```

Comment:
Which of these is the base case (line 171 before the change) and which ones are specific to the 71.2 issue?

Reply:
Line 171 before the change is the root cause of the problem, so we shouldn't let it stay the same. The fix is to make sure 71.3 and 71.2 use the same de/serialization order so that the segment fault or some weird deserialized result won't appear.